guide/additional-features/cooldowns.md

```diff
@@ -0,0 +1,87 @@
+# Cooldowns
+
+Spam is something you generally want to avoid, especially if one of commands require calls to other APIs or takes a bit of time to build/send.
+
+This section assumes you followed the [Command Handling](/guide/creating-your-bot/command-handling.md) part.
+
+First, add a cooldown property to your command. This will determine how long the user would have to wait (in seconds) before using the command again.
+
+```js {4}
+const { SlashCommandBuilder } = require('discord.js');
+module.exports = {
+	cooldown: 5,
+	data: new SlashCommandBuilder()
+		.setName('ping')
+		.setDescription('Replies with Pong!'),
+	async execute(interaction) {
+		// ...
+	},
+};
+```
+
+In your main file, initialize a [Collection](/guide/additional-info//collections.md) to store cooldowns of commands
+
+```js {1}
+client.cooldowns = new Collection();
+```
+
+The key will be the command names, and the values will be Collections associating the user's id (key) to the last time (value) this specific user used this specific command. Overall the logical path to get a specific user's last usage of a specific command will be `cooldowns > command > user > timestamp`.
+
+In your main file, add the following code:
+
+```js {1,3-5,7-10,12-14}
+const { cooldowns } = client;
+if (!cooldowns.has(command.data.name)) {
+	cooldowns.set(command.data.name, new Collection());
+}
+const now = Date.now();
+const timestamps = cooldowns.get(command.data.name);
+const defaultCooldownDuration = 3;
+const cooldownAmount = (command.cooldown || defaultCooldownDuration) * 1000;
+if (timestamps.has(interaction.user.id)) {
+	// ...
+}
+try {
+	// ...
+} catch (error) {
+	// ...
+}
+```
```

Comment:
There should be new lines before and after the `if`s and `try...catch`. The highlights seem to be incorrect also.